src/utils/types.zig

```diff
@@ -264,3 +264,74 @@ pub inline fn comptimeZeroSizePtrCast(comptime T: type, comptime ptr: *const any
         return dummy.value;
     }
 }
+
+// config: allocator, pointers
+
+/// Compare equality of two items with the same type.
+pub fn eql(one: anytype, two: anytype) bool {
```

Comment:
isnt this the same as `std.meta.eql` ?

Reply:
The key area where this differs is that it follows slices to compare the data they point to, instead of comparing the pointers themselves for equality. In general I think when comparing equality, obviously there are plenty of exceptions, but I think typically you want to compare what's behind the pointer, not compare the pointer itself. 

This method needs a bit more time spent on it to flesh it out properly. I started messing with it but it was taking too much time. But I'd like to revisit it at some point.

Reply:
ok yeah thats a great addition then - can add more comments/docs to it later